app/payment-link-v2/confirm/confirm.controller.test.js

```diff
@@ -190,4 +196,92 @@ describe('Confirm Page Controller', () => {
+        type: 'ADHOC',
+        reference_label: 'Invoice number',
+        reference_enabled: true,
+        price: 1000,
+        require_captcha: true
+      }))
+
+      it('when a failed captcha is entered, it should display an error and ' +
+        'redirect to the confirm page and show the values from the hidden form fields to display ' +
+        'the page and prevent session interference', async () => {
+        req = {
+          correlationId: '123',
+          product,
+          body: {
+            'reference-value': 'a-invoice-number',
+            amount: '10.00',
+            'g-recaptcha-response': 'recaptcha-test-token'
+          }
+        }
+
+        res = {
+          locals: {
+            __p: sinon.stub()
+          }
+        }
+
+        mockCaptcha.verifyCAPTCHAToken.resolves(false)
+
+        res.locals.__p.withArgs('paymentLinksV2.confirm.totalToPay').returns('Total to pay')
+        res.locals.__p.withArgs('paymentLinksV2.fieldValidation.youMustSelectIAmNotARobot')
+          .returns('You failed the captcha challenge.  Please try again.')
+
+        await controller.postPage(req, res)
+
+        sinon.assert.calledWith(responseSpy, req, res, 'confirm/confirm')
+
+        const pageData = mockResponses.response.args[0][3]
+        expect(pageData.summaryElements.length).to.equal(2)
+        expect(pageData.summaryElements[0].summaryLabel).to.equal('Invoice number')
+        expect(pageData.summaryElements[0].summaryValue).to.equal('a-invoice-number')
+        expect(pageData.summaryElements[1].summaryLabel).to.equal('Total to pay')
+        expect(pageData.summaryElements[1].summaryValue).to.equal('£10.00')
+
+        expect(pageData.errors).to.contain({
+          recaptcha: 'You failed the captcha challenge.  Please try again.'
+        })
+      })
+
+      it('when the captcha call fails, it should display an error and ' +
+        'redirect to the confirm page', async () => {
+        req = {
+          correlationId: '123',
+          product,
+          body: {
+            'reference-value': 'a-invoice-number',
+            amount: '10.00',
+            'g-recaptcha-response': 'recaptcha-test-token'
+          }
+        }
+
+        res = {
+          locals: {
+            __p: sinon.stub()
+          }
+        }
+
+        mockCaptcha.verifyCAPTCHAToken.rejects()
+
+        res.locals.__p.withArgs('paymentLinksV2.confirm.totalToPay').returns('Total to pay')
+        res.locals.__p.withArgs('paymentLinksV2.fieldValidation.youMustSelectIAmNotARobot')
+          .returns('There was an issue with the recaptcha.  Please try again.')
+
+        await controller.postPage(req, res)
+
+        sinon.assert.calledWith(responseSpy, req, res, 'confirm/confirm')
+
+        const pageData = mockResponses.response.args[0][3]
+
+        expect(pageData.errors).to.contain({
+          recaptcha: 'There was an issue with the recaptcha.  Please try again.'
+        })
```

Comment:
should we add in checks that the pagedata for for the summary is sent correctly?

Reply:
What I did was merge that check into the earlier - so I called it:

```
when a failed captcha is entered, it should display an error and ' +
        'redirect to the confirm page and show the values from the hidden form fields to display ' +
        'the page and prevent session interference
```

https://github.com/alphagov/pay-products-ui/pull/1766/files/9d5ec47827f99ae8ae129009bf7589c9a3ea184e#diff-db5847a93661822df790c16ecbccf290ceee0ec4dadaf0d27bd9f5203e667249R210

Reply:
Just pushed a change to fix the grammar:

```
it('when a failed captcha is entered, it should display an error and ' +
'redirect to the confirm page and show the values from the hidden form fields to display ' +
'the page so that it prevents session interference', async () => 
```

Please let me know if you think I should separate out that bit fo the test.